thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -321,12 +322,14 @@ namespace diskann {
     std::vector<uint64_t> tmp_result_ids_64(sample_num, 0);
     std::vector<float>    tmp_result_dists(sample_num, 0);
 
+    this->use_sample = true;
```

Comment:
Can you explain why we need this?

Also I don't think it is a good idea to add it to the param list of `cached_beam_search` since the API need to be as lite as possible. This one can be a field of the class since no one outside need to set it right?

Reply:
> Can you explain why we need this?
> 
> Also I don't think it is a good idea to add it to the param list of `cached_beam_search` since the API need to be as lite as possible. This one can be a field of the class since no one outside need to set it right?

became when diskann generates cache nodes, it used sample vector data in the build stage. when using IP metric, diskann will add an extra dim, like 256(original dim) -> 257(L2 norm add a dim) -> 264(aligned dim). but query vector data have not been L2 normed, it's dim like 256 -> 256 (aligned dim).

yes, I move the sample field from `cache_beam_search`, and use an extra flag to indicate generate cache stage. the same idea also see  [diskann](https://github.com/microsoft/DiskANN/blob/main/src/pq_flash_index.cpp#L293)

Reply:
Do you mind to add comment about this here. Or it is hard to understand